cookbooks/finite_strain/doc/finite_strain.cc

```diff
@@ -18,17 +18,18 @@
   <http://www.gnu.org/licenses/>.
 */
 
-for (unsigned int q=0; q < in.position.size(); ++q)
+for (unsigned int q=0; q < in.n_evaluation_points(); ++q)
   {
     // Convert the compositional fields into the tensor quantity they represent.
-    Tensor<2,dim> strain;
-    for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
-      strain[Tensor<2,dim>::unrolled_to_component_indices(i)] = in.composition[q][i];
+    const ArrayView<const double> strain_values(&in.composition[q][0],
+                                                Tensor<2,dim>::n_independent_components);
+    const Tensor<2,dim> strain(strain_values);
 
     // Compute the strain accumulated in this timestep.
     const Tensor<2,dim> strain_increment = this->get_timestep() * (velocity_gradients[q] * strain);
 
     // Output the strain increment component-wise to its respective compositional field's reaction terms.
-    for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
-      out.reaction_terms[q][i] = strain_increment[Tensor<2,dim>::unrolled_to_component_indices(i)];
+    const ArrayView<double> strain_increment_values(&out.reaction_terms[q][0],
+                                                    Tensor<2,dim>::n_independent_components);
+    strain_increment.unroll(strain_increment_values.begin(), strain_increment_values.end());
```

Comment:
wouldn't be simpler to do
```suggestion
    strain_increment.unroll(strain_increment.unroll(&out.reaction_terms[q][0], &out.reaction_terms[q][0]+Tensor<2,dim>::n_independent_components);
```
?

Reply:
I am also not sure why ``unroll`` actually needs an end iterator. The interface would be nicer if it would take a single iterator...